### ISSUE TITLE
transform insert at the same position - incorrect result

### ISSUE THREAD

**Issue by gkubisa** [ottypes/rich-text#25]
As I understand, when the `side` param to `transform` is "left", the first operation should "happen" first. However, the logic here seems to be reversed.

```
> require('rich-text').type.transform([ { insert: 'a' } ], [ {insert: 'b'} ], 'left')
Delta { ops: [ { retain: 1 }, { insert: 'a' } ] } // I'd expect Delta { ops: [ { insert: 'a' } ] }
```

For comparison:

```
> require('ot-text').type.transform([ 'a' ], [ 'b' ], 'left')
[ 'a' ] // as expected
```

I think [this line](https://github.com/ottypes/rich-text/blob/master/lib/type.js#L36) should be changed to `return delta2.transform(delta1, side === 'right');`

**Comment by jhchen** [comment]
> when the side param to transform is "left", the first operation should "happen" first

Do you have a reference for this? The API spec just specifies it how it is used (tiebreakers) and that `apply(apply(snapshot, op1), transform(op2, op1, 'left')) == apply(apply(snapshot, op2), transform(op1, op2, 'right'))` which is does.

**Comment by gkubisa** [comment]
Thanks for the quick reply @jhchen. The spec is very vague wrt to the `side` param, so no, I don't have a reference. I raised this issue only because the behaviour is inconsistent between [rich-text](https://github.com/ottypes/rich-text) and [text](https://github.com/ottypes/text) and intuitively the latter (text) seems right.

I'd think that `apply(apply(snapshot, op1), transform(op2, op1, 'left')) == apply(apply(snapshot, op2), transform(op1, op2, 'right'))` would also be satisfied, if the `side` was "swapped" as I suggested. So, it's your call, if you want to make this change.
